### PR TITLE
persist and pass jwt

### PR DESCRIPTION
**Pull request by achiu8**
https://github.com/yujinjcho/rcruitme-front/issues/18

### CODE REVIEW

File: src/formFrom.js [hunk 1]
```diff
@@ -39,25 +40,26 @@ export default formFields => {
       fetch(this.props.endpoint, {
         method: 'POST',
         headers: {
+          ...auth.header(),
```
Comment (yujinjcho) [inline src/formFrom.js]: Does this includes the header on `sign-up` too? which is why I was seeing [this](https://github.com/yujinjcho/rcruitme/pull/38#issuecomment-455291418)?

Comment (achiu8) [reply src/formFrom.js]: Yeah, includes for `sign-up` post too, but shouldn't result in the error you were seeing..

So actually I realized this branch doesn't include [these changes](https://github.com/yujinjcho/rcruitme-front/pull/17), and I had just pulled them in locally to test. If I remove them and go strictly off this branch, sign-up doesn't work for me either, but for a different reason (missing `redirect` param). Are you sure that's not actually why failing for you? The other error you're seeing might be left over from before you cleared cookies?

**Comment by achiu8** [comment]
Okay I think maybe a few issues..
* Wasn't passing `redirect` on sign-up request - merged master
* Shouldn't pass auth token for sign-in or sign-up - now set based on new `authed` prop in form
* CSRF filter - added `X-Requested-With` header, along with new filter config on server

**Comment by yujinjcho** [comment]
I'm a little confused. 

If I exclude the following from `application.conf`
```
play.filters.csrf.header.bypassHeaders {
  X-Requested-With = "*"
}
```

 and also comment out [this line](https://github.com/yujinjcho/rcruitme-front/pull/19/files#diff-690bc58469eaade65261d9309412ba9bR44) (don't send the `X-Requested-With` header), then a sign-up shouldn't work?  Seems like it does work though..

If I add the bypassHeader config back into the `application.conf`, a sign-up still works without the header. 

I guess I'm not too sure what is happening.

**Comment by yujinjcho** [comment]
I mean things seem to be working. I'd be okay merging all the branches and trying to figure out afterwards what we may or may not need afterwards. kind of confusing having 3 branches outstanding.

**Comment by achiu8** [comment]
Sign-up won't work if for some reason you are already logged in or have some other thing stored in `localStorage` at `token` key. I mean unlikely but either way that request shouldn't be authed / made with credentials.

**Comment by yujinjcho** [comment]
Just trying to understand, but shouldn't the sign-up still be working as long as you have the `X-Requested-With` header even if you're signed in?

**Comment by achiu8** [comment]
Oh whoops was kinda drunk, was thinking about the auth header..

For some reason any other cookies seem to cause the CSRF filter to fail. Adding the `X-Requested-With` header and config will bypass that filter completely. Without them, you'd first have to clear cookies before any post requests work.

**Pull request merged**